x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/document_detail_logic.ts

```diff
@@ -65,13 +67,17 @@ export const DocumentDetailLogic = kea<DocumentDetailLogicType>({
 
       try {
         const { http } = HttpLogic.values;
-        // TODO: Handle 404s
         const response = await http.get(
           `/api/app_search/engines/${engineName}/documents/${documentId}`
         );
         actions.setFields(response.fields);
       } catch (e) {
-        flashAPIErrors(e);
+        // If an error occurs trying to load this document, it will typically be a 404, or some other
+        // error that will prevent the page from loading, so redirect to the documents page and
+        // show the error
+        flashAPIErrors(e, { isQueued: true });
```

Comment:
haha I know I wrote this helper myself but my god I definitely completely forgot that I added an `isQueued` flag until just now. Good job, past Constance 👵